chain/index/msgindex.go

```diff
@@ -88,7 +89,7 @@ type headChange struct {
 	app []*types.TipSet
 }
 
-func NewMsgIndex(lctx context.Context, path string, cs ChainStore) (MsgIndex, error) {
+func NewMsgIndex(lctx context.Context, path string, cs ChainStore, enableWrites bool) (MsgIndex, error) {
```

Comment:
`enableWrites` is just a temporary thing until this module gets removed entirely?

Reply:
ah, I see "If chainIndexer fails, fallback to msgIndex" below; we're maintaining both until we come up with a migration strategy? Or this is part of the migration strategy? I can imagine this being a gentle step in the process but at some point we need to pull a harder trigger.

Reply:
Have now completely got rid of the `MsgIndex`. Dual reads make it hard to reason about what's contained in the Index and will need us to maintain the old Indices for the time being. Let's just rip the band aid off in one go by putting a robust migration strategy in place.